### PR TITLE
test: Check the equality between two type references.

### LINKED ISSUE #317: TypeReference.equals is wrong
For generic type arguments, TypeReference("C extends Foo") and TypeReference("C") are equals. They should not be.

### PR DESCRIPTION
**Pull request by GerardPaligot**
Closes #317

### CODE REVIEW

File: src/test/java/spoon/test/reference/TypeReferenceTest.java [hunk 1]
```diff
@@ -166,11 +156,35 @@ public void testNullReferenceSubtype() throws Exception {
...
+		final Launcher launcher = new Launcher();
+		launcher.setArgs(new String[] {"--output-type", "nooutput", "--noclasspath"});
+		launcher.addInputResource("src/test/java/spoon/test/reference/TypeReferenceTest.java");
+		launcher.run();
+
+		Factory factory = launcher.getFactory();
+		final CtTypeReference<?> firstRef = factory.Type().get(A.Tacos.class).getFormalTypeParameters().get(0);
+		final CtTypeReference<?> secondRef = factory.Type().get(B.Tacos.class).getFormalTypeParameters().get(0);
+
+		assertNotEquals(firstRef.toString(), secondRef.toString());
```
Comment (monperrus) [inline src/test/java/spoon/test/reference/TypeReferenceTest.java]: + two assertions
assertNotEquals(firstRef, secondRef); 
assertNotEquals(firstRef.getSignature(), secondRef.getSignature());

Comment (GerardPaligot) [inline src/test/java/spoon/test/reference/TypeReferenceTest.java]: Done.

File: src/test/java/spoon/test/reference/TypeReferenceTest.java [hunk 2]
```diff
@@ -166,11 +158,36 @@ public void testNullReferenceSubtype() throws Exception {
...
+	@Test
+	public void unboxTest() {
+		Factory factory = new Launcher().createFactory();
+		CtTypeReference<Boolean> boxedBoolean = factory.Class().createReference(Boolean.class);
+		assertEquals(boxedBoolean.unbox().getActualClass(), boolean.class);
+	}
+
+	@Test
+	public void testToStringEqualityBetweenTwoGenericTypeDifferent() throws Exception {
+		final Launcher launcher = new Launcher();
```
Comment (monperrus) [inline src/test/java/spoon/test/reference/TypeReferenceTest.java]: contract: generic type references with different bounds should not be considered equals

**Comment by GerardPaligot** [comment]
@monperrus Travis ok.

**Pull request merged**